smithy-cli/src/main/java/software/amazon/smithy/cli/dependencies/MavenDependencyResolver.java

```diff
@@ -96,11 +96,11 @@ public void serviceCreationFailed(Class<?> type, Class<?> impl, Throwable except
     public void addRepository(MavenRepository repository) {
         try {
             URI uri = new URI(repository.getUrl());
-            String name = uri.getHost();
+            String id = repository.getId().orElseGet(() -> uri.getHost());
```

Comment:
**suggestion:** now that I think of it, perhaps we could default to the index instead (in the list of available repos)? That way it's still going to be unique.

Reply:
That should work. Maybe something like `index + "-" + uri.getHost()` for better logs/debugging?

Reply:
Done. 
Index starts at 1, let me know if you want to bikeshed 😄